doc/neps/ufunc-overrides.rst

```diff
@@ -575,14 +591,78 @@ For option (2), the relevant example is the fourth, with ``arr *=
+are nearly identical, in most cases options (1) and (2) will
+yield the same result with the same implementation of ``__array_ufunc__``.
+One exception is the order in which implementations are tried when the
+second argument is a subclass of the first argument, due to a Python
+bug [9]_ expected to be fixed in Python 3.7.
+
+In general, we recommend adopting option (1), which is the option most
+similar to that used by :class:`ndarray` itself. Note that option (1)
+is viral, in the sense that any other class that wishes to support binary
+operations with your class now must also follow these rules for supporting
+binary arithmetic with :class:`ndarray` (i.e., they must either implement
+``__array_ufunc__`` or set it to :obj:`None`). We believe this is a good
+thing, because it ensures the consistency of ufuncs and arithmetic on all
+objects that support them.
+
+To make implementing such array-like classes easier, the mixin class
+:class:`~numpy.lib.mixins.NDArrayOperatorsMixin` provides option (1) style
+overrides for all binary operators with corresponding Ufuncs. Classes
+that wish to implement ``__array_ufunc__`` for compatible versions
+of NumPy but that also need to support binary arithmetic with NumPy arrays
+on older versions, should ensure that ``__array_ufunc__`` can also be used
+to implement all binary operations they support.
+
 Finally, we note that we had extensive discussion about whether it might
 make more sense to ask classes like ``MyObject`` to implement a full
 ``__array_ufunc__`` [6]_. In the end, allowing classes to opt out was
 preferred, and the above reasoning led us to agree on a similar
-implementation for :class:`ndarray` itself. To help implement array-like
-classes, the mixin class :class:`~numpy.lib.mixins.NDArrayOperatorsMixin`
-provides overrides for all binary operators with corresponding ufuncs.
-
+implementation for :class:`ndarray` itself. We decided to require disabling
+ufuncs to opt out to ensure that a class cannot define Ufuncs to return
+different results than the corresponding binary operations (i.e., if
+``np.add(x, y)`` is defined, it should match ``x + y``).
+
+.. [9] http://bugs.python.org/issue30140
+
+.. _neps.ufunc-overrides.list-operations:
+
+List of binary operations and NumPy Ufuncs
+------------------------------------------
+
+Here is a full list of Python binary operations and the corresponding NumPy
+Ufuncs used by :class:`ndarray`:
+
+====== ============ =========================================
+Symbol Operator     NumPy Ufunc(s)
+====== ============ =========================================
+``<``  ``lt``       :func:`less`
+``<=`` ``le``       :func:`less_equal`
+``==`` ``eq``       :func:`equal`
+``!=`` ``ne``       :func:`not_equal`
+``>``  ``gt``       :func:`greater`
+``>=`` ``ge``       :func:`greater_equal`
+``+``  ``add``      :func:`add`
+``-``  ``sub``      :func:`subtract`
+``*``  ``mul``      :func:`multiply`
+``/``  ``truediv``  :func:`true_divide`
+       (Python 3)
```

Comment:
Sphinx does a weird thing if I put it on one line, making the column very wide (I think this is basic on the longer length of the column header `======`). So this version ends up looking better when rendered to HTML.